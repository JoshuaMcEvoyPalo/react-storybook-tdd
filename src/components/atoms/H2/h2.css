.h2 {
    font-family: 'Exo', serif;
    font-style: normal;
    font-weight: 800;
    font-size: 24px;
    line-height: 120%;
    color: #112D4E;
}