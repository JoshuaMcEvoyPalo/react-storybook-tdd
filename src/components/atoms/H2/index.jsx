import React from 'react'
import './h2.css'

const H2 = ({ children }) => {
  return (
    <h2 data-testid="h2" className="h2">{children}</h2>
  )
}

export default H2
