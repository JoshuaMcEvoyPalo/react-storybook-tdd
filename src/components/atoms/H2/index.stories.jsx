import H2 from '.'

// storiesOf('Atoms / H2', module).add('Default', () => {
//   <H2>{"H2 Title"}</H2>
// })

// SETUP
export default {
  title: 'Atoms / H2',
  component: H2,
}

// TEMPLATE
const Template = (args) => <H2 {...args} />

// STORIES VARIANTS
export const Default = Template.bind({})
Default.args = {
  children: "Title Heading",
}
