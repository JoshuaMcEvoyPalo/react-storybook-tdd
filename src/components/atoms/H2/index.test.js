import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer';
import H2 from '.'

describe('H2 tests', () => {
  const title = "Heading text"
  it('Should render H2 correctly', () => {
    const tree = renderer.create(<H2>{title}</H2>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Should render children', () => {
    render(<H2>{title}</H2>)
    expect(screen.getByTestId('h2')).toContainHTML(title)
  })
})