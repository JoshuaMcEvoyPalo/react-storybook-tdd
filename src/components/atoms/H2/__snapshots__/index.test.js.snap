// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H2 tests Should render H2 correctly 1`] = `
<h2
  className="h2"
  data-testid="h2"
>
  Heading text
</h2>
`;
